bjoern/request.c

```diff
@@ -205,7 +205,18 @@ on_message_complete(http_parser* parser)
   /* SERVER_NAME and SERVER_PORT */
   if (REQUEST->server_info->host) {
     _set_header(_SERVER_NAME, REQUEST->server_info->host);
-    _set_header(_SERVER_PORT, REQUEST->server_info->port);
+    if (REQUEST->server_info->port == Py_None) {
+      _set_header(_SERVER_PORT,
+                  PyUnicode_FromFormat(""));
+    } else {
```

Comment:
In theory neither SERVER_NAME nor SERVER_PORT must be empty. Should a sane default of "localhost" be used? HTTP_HOST should take precedence in WSGI apps. Port 80 may be used  with used domain sockets (with unix sockets the reconstructed url is wrong in every case!).